starport/services/network/request.go

```diff
@@ -0,0 +1,30 @@
+package network
+
+import (
+	"context"
+
+	launchtypes "github.com/tendermint/spn/x/launch/types"
+	"github.com/tendermint/starport/starport/pkg/yaml"
+)
+
+// FetchRequest fetches the chain request from SPN by launch and request id
+func (n Network) FetchRequest(ctx context.Context, launchID, requestID uint64) (string, error) {
```

Comment:
I don't see why we convert to yaml here and why we return a string. Representing data differently should be done in the client side I think, we should return the Request object directly here.

Also, why do we need to encode in yaml format to render it on terminal?

Reply:
The Request object can be tricky to parse all types with all parameters. Converting to YAML is an easy way. The content can be more them five different types, and some content fields use a byte array instead of a string, like the pubkey into the`RequestContent_GenesisValidator` type. I converted to YAML to have a human-readable output. Do you have any other ideas on it?

Reply:
I still believe that the Request func should not return a string, it should return an object like any other query commands we have do. If we need to represent the content in yaml (looks good to me), this should be done in the client side (cmd).

Reply:
sure, it make sense, also I need to reuse the `Request` method, I will move this logic to the cmd